### ISSUE TITLE
Import zfl is ridiculously slow

### ISSUE THREAD

**Issue by Michaelwhite34** [funexpected/godot-flash-module#13]
Should add an option not to compress to zip and directly use folder structure.

**Comment by Michaelwhite34** [comment]
No, I guess it's the problem of the module. In the import. folder, the extracted folder finishes within 1 min and it is not finished after nearly 1 hour.

**Comment by ky000** [comment]
> No, I guess it's the problem of the module. In the import. folder, the extracted folder finishes within 1 min and it is not finished after nearly 1 hour.



you can try to rasterize the library to increase the speed. Here it takes some time, but my library is very big, And yes, the extracted folder usually ends up first. And even when adobe crashes, the .zfl file is still being created. He walks all over the timeline and everything. It's a well-crafted process.

**Comment by Michaelwhite34** [comment]
> > No, I guess it's the problem of the module. In the import. folder, the extracted folder finishes within 1 min and it is not finished after nearly 1 hour.
> 
> you can try to rasterize the library to increase the speed. Here it takes some time, but my library is very big, And yes, the extracted folder usually ends up first. And even when adobe freeze/crash the .zfl file is still being created. He walks all over the timeline and everything.

It's alright after disabling ETC2, but the compression is really unnecessary.

**Comment by Michaelwhite34** [comment]
Should add an option to disable s3tc too. Because when I turn that off in godot, the animation won't show.

**Comment by Michaelwhite34** [comment]
> > No, I guess it's the problem of the module. In the import. folder, the extracted folder finishes within 1 min and it is not finished after nearly 1 hour.
> 
> you can try to rasterize the library to increase the speed. Here it takes some time, but my library is very big, And yes, the extracted folder usually ends up first. And even when adobe freeze/crash the .zfl file is still being created. He walks all over the timeline and everything.

The main problem for me at the moment is adobe animate freezes when exporting a bit larger file with flash-tools, I left it open overnight and it does nothing.

**Comment by Michaelwhite34** [comment]
Another thing I am not sure If I should post it in the repo, some parts of the animation doesn't have the right color in godot. Don't know if it's the problem of the export or displaying in godot.

**Comment by ky000** [comment]
> Another thing I am not sure If I should post it in the repo, some parts of the animation doesn't have the right color in godot. Don't know if it's the problem of the export or displaying in godot.

The colors look good to me. Something like this happened when my Graphic had an instance applying Filter and another duplicate instance applying another filter. Turns out the filters matched and the final color was different. After correcting and making only one instance applying the Color Filter the problem was solved. But I haven't tested that much.

**Comment by Michaelwhite34** [comment]
> > Another thing I am not sure If I should post it in the repo, some parts of the animation doesn't have the right color in godot. Don't know if it's the problem of the export or displaying in godot.
> 
> The colors look good to me. Something like this happened when my Graphic had an instance applying Filter and another duplicate instance applying another filter. Turns out the filters matched and the final color was different. After correcting and making only one instance applying the Color Filter the problem was solved. But I haven't tested that much.

Is there any documentation about how to use Flashplayer in gdscript ?

**Comment by ky000** [comment]
> > > Another thing I am not sure If I should post it in the repo, some parts of the animation doesn't have the right color in godot. Don't know if it's the problem of the export or displaying in godot.
> > 
> > 
> > The colors look good to me. Something like this happened when my Graphic had an instance applying Filter and another duplicate instance applying another filter. Turns out the filters matched and the final color was different. After correcting and making only one instance applying the Color Filter the problem was solved. But I haven't tested that much.
> 
> Is there any documentation about how to use Flashplayer in gdscript ?

I'm not the best person to help you with that. I recommend that you submit a documentation request. I have also submitted one for assistance with changing colors in separate symbols, but I haven't received a response yet

**Comment by Michaelwhite34** [comment]
> > > > Another thing I am not sure If I should post it in the repo, some parts of the animation doesn't have the right color in godot. Don't know if it's the problem of the export or displaying in godot.
> > > 
> > > 
> > > The colors look good to me. Something like this happened when my Graphic had an instance applying Filter and another duplicate instance applying another filter. Turns out the filters matched and the final color was different. After correcting and making only one instance applying the Color Filter the problem was solved. But I haven't tested that much.
> > 
> > 
> > Is there any documentation about how to use Flashplayer in gdscript ?
> 
> I'm not the best person to help you with that. I recommend that you submit a documentation request. I have also submitted one for assistance with changing colors in separate symbols, but I haven't received a response yet

After several hours' experiment, I figured out several codes that I need to use. Just looked up in flash_player.cpp.
